src/source/Ice/IceAgentStateMachine.c

```diff
@@ -405,6 +405,15 @@ STATUS executeGatheringIceAgentState(UINT64 customData, UINT64 time)
 
     CHK_LOG_ERR_NV(retStatus);
 
+    if (STATUS_FAILED(retStatus)) {
+        pIceAgent->iceAgentStatus = retStatus;
+        // step into failed state
+        stepStateMachine(pIceAgent->pStateMachine);
+
+        // fix up retStatus so we can successfully transition to failed state.
+        retStatus = STATUS_SUCCESS;
```

Comment:
So only `STATUS_SUCCESS` is possible? Is it worth having these functions return a status if they don't mean anything?

Reply:
the state machine is a pic construct. The way it works is that if any of the state callbacks return fail status then it bails immediately. But I want it to move to failed state instead, so I have to store the error status in pIceAgent->iceAgentStatus and  fix up the retStatus.

Reply:
Aye, ok will approve!